### DISCUSSION TITLE
Multiple HA instances or Multiple Tunnels.

### DISCUSSION THREAD

**Discussion by sambarlick** [homeassistant-apps/app-cloudflared#381] [Q&A] [answered]
Hi,

I was wondering if it possible to have one Home Assistant CloudFlared add-on create tunnels for multiple Home Assistant instances. Or can I have the CloudFlared add-on installed on multiple HA instances linked to the same domain?

Thanks.

**Comment by brenner-tobias** [comment, marked as answer]
Hi,
sure, you can just use one HA with one tunnel and add the other HA instance in the additional_hosts array. In that scenario, all you have to adapt is the entry in the http integration of the second HA instance, since you do not have to enter 

```
  trusted_proxies:
    - 172.30.33.0/24
```

but something like

```
  trusted_proxies:
    - 192.168.1.10
```

where "192.168.1.10" would be the IP of your first HA running Cloudflared. This is the case, since your second HA has to allow for proxied connections from the first one.

The other option is also working: Simply install Cloudflared on both instances and link both to the same account. You will then have two tunnels. All you have to make sure is to use different tunnel_names and external_hostnames.

Let me know if anything was not clear or if you have any additional questions.
